pages/turing_test_page.py

```diff
@@ -53,69 +116,136 @@
-    fig.update_layout(margin=dict(l=20, r=20, t=20, b=20),
-                      paper_bgcolor="LightSteelBlue")
-    fig.update_layout(coloraxis_showscale=False)
-    fig.update_xaxes(showticklabels=False)
-    fig.update_yaxes(showticklabels=False)
-    return fig, image_type[t]
+
+def answer_is_correct(button_id, img_t):
+    return button_id == 'true-button' and img_t == image_type[1] \
+           or button_id == 'false-button' and img_t == image_type[0]
+
+
+@callback(
+    Output("true-button", "disabled"),
+    Output("false-button", "disabled"),
+    Output("dummy-div", "hidden"),
+    Input("browser-storage", "data"),
+    prevent_initial_call=True
+)
+def toggle_buttons(storage_data):
+    if storage_data['current_que'] < MAX_QUESTIONS:
+        return False, False, True
+    else:
+        return True, True, False
 
 
 @callback(
     Output("success-bar", "value"),
+    Output("success-bar", "label"),
     Output("fail-bar", "value"),
+    Output("fail-bar", "label"),
     Output("image-graph", "figure"),
-    Output("image-type", "children"),
+    Output("browser-storage", "data"),
     Input("true-button", "n_clicks"),
     Input("false-button", "n_clicks"),
+    Input("reset-test-button", "n_clicks"),
     Input("success-bar", "value"),
     Input("fail-bar", "value"),
-    Input("image-type", "children")
+    Input("browser-storage", "data"),
+    prevent_initial_call=True
 )
-def true_button_clicked(n_s, n_f, val_s, val_f, img_t):
+def some_button_clicked(_n_s, _n_f, _n_r, val_s, val_f, storage_data):
     ctx = dash.callback_context
 
     if not ctx.triggered:
         button_id = 'No clicks yet'
     else:
         button_id = ctx.triggered[0]['prop_id'].split('.')[0]
 
-    if button_id == 'true-button' and img_t == image_type[1] or button_id == 'false-button' and img_t == image_type[0]:
-        val_s += 1
-    else:
-        val_f += 1
-    return val_s, val_f, *create_image()
-
-# @callback(
-#     Output("fail-bar", "value"),
-#     Input("false-button", "n_clicks"),
-#     Input("fail-bar", "value"),
-# )
-# def false_button_clicked(n, val):
-#     return val + 1
+    if button_id in ['true-button', 'false-button']:
+        img_i = storage_data['current_que']
+        img_t = storage_data['questions'][img_i]['label']
+
+        if answer_is_correct(button_id, img_t):
+            val_s += 1
+            answer = True
+            storage_data['answers'].append(answer)
+        else:
+            val_f += 1
+            answer = False
+            storage_data['answers'].append(answer)
+
+        storage_data['current_que'] += 1
+
+        img_i = storage_data['current_que']
+        if img_i < MAX_QUESTIONS:
+            n_img = storage_data['questions'][img_i]['img']
+        else:
+            n_img = storage_data['questions'][-1]['img']
+            # TODO save answers
+            # print(storage_data['answers'])
+            return val_s, str(val_s), val_f, str(val_f), n_img, storage_data
```

Comment:
Czemu nie możesz przypisać `None` albo coś? Jaką długość ma lista/tablica `questions`, i czemu na końcu jest jakiś specjalny element, i czemu nie możesz tego pobrać po prostu indeksem `MAX_QUESTIONS`?

Reply:
None wysra widget chyba, a `arr[-1]` to w pythonie normlane odniesienie się do ostatniego elementu, a daje ostatni, zeby było wrażenie zamrożenia

Reply:
Ok, jest to jakieś uzasanienie, ale ja bym po prostu zrobił

```suggestion
        n_img = storage_data['questions'][img_i]['img']
        if img_i == MAX_QUESTIONS:     
            # TODO save answers
            # print(storage_data['answers'])
            return val_s, str(val_s), val_f, str(val_f), n_img, storage_data
```